docs/templates/query_vs_dataframe.md

```diff
@@ -0,0 +1,70 @@
+# Query vs DataFrame
+
+HSFS provides a DataFrame API to ingest data into the Hopsworks Feature Store. You can also retrieve feature data in a DataFrame, that can either be used directly to train models or [materialized to file(s)](training_dataset.md) for later use to train models.
+
+The idea of the Feature Store is to have pre-computed features available for both training and serving models. The key functionality required to generate training datasets from reusable features are: feature selection, joins, filters and point in time queries. To enable this functionality, we are introducing a new expressive Query abstraction with `HSFS` that provides these operations and guarantees reproducible creation of training datasets from features in the Feature Store.
+
+The new joining functionality is heavily inspired by the APIs used by Pandas to merge DataFrames. The APIs allow you to specify which features to select from which feature group, how to join them and which features to use in join conditions.
+
+```python
+# create a query
+feature_join = rain_fg.select_all()
+                         .join(temperature_fg.select_all(), on=["date", "location_id"])
```

Comment:
I'd keep this first example simple and also mention how the joining key is selected.

Reply:
added more examples also with scala equivalents.